exercise/static/exercise/results_staff.js

```diff
@@ -705,9 +705,30 @@
         }
     }
 
+    let ajaxRetryCount = 0;
+	let ajaxRetryLimit = 10;
+
+	let ajaxSettings = {
+	  timeout: 10000,
+      beforeSend: function(xhr, settings) {
+        return ajaxRetryCount <= ajaxRetryLimit;
+      },
+	  error: function(xhr, statusText, errorThrown) {
+		ajaxRetryCount++;
+		if (ajaxRetryCount <= ajaxRetryLimit) {
+		  console.log("Retrying ajax:", statusText);
+		  $.ajax(this);
+		  return;
+		} else {
+          $("#ajax-failed-alert").show();
+          $("#results-loading-animation").hide(); 
+        }
+	  }
+	};		  
+
     function gatherFromAPIPaging(url) {
         function gather(cur_result, url) {
-            return $.ajax(url)
+            return $.ajax({...ajaxSettings, url: url})
```

Comment:
Is this syntax supported by internet explorer 11 or Edge (new in ECMAScript 2018). Should be replaced with `$.extend({}, ajaxSettings, {url: url})`. ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Browser_compatibility

Reply:
We do have lots of JS code that does not work in IE 11, but if the latest Edge does not work either, then that's bad.